src/components/structures/RoomDirectory.tsx

```diff
@@ -116,7 +119,30 @@ export default class RoomDirectory extends React.Component<IProps, IState> {
         } else if (!selectedCommunityId) {
             MatrixClientPeg.get().getThirdpartyProtocols().then((response) => {
                 this.protocols = response;
-                this.setState({ protocolsLoading: false });
+                const myHomeserver = MatrixClientPeg.getHomeserverName();
+                const lsRoomServer = localStorage.getItem(LAST_SERVER_KEY);
+                const lsInstanceId = localStorage.getItem(LAST_INSTANCE_KEY);
+                const configSevers = SdkConfig.get().roomDirectory?.servers || [];
+                const settingsServers = SettingsStore.getValue("room_directory_servers") || [];
+                const roomServer = [...configSevers, ...settingsServers].includes(lsRoomServer)
+                    ? lsRoomServer
+                    : myHomeserver;
```

Comment:
```suggestion

                const roomServer = (
                	SdkConfig.get().roomDirectory?.servers?.includes(lsRoomServer) ||
                	SettingsStore.getValue("room_directory_servers")?.includes(lsRoomServer)
                ) ? lsRoomServer : myHomeserver;
```

or similar is more readable IMO